lib/ansible/modules/clustering/kubevirt/kubevirt_template.py

```diff
@@ -0,0 +1,271 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2018, KubeVirt Team <@kubevirt>
+# Apache License, Version 2.0
+# (see LICENSE or http://www.apache.org/licenses/LICENSE-2.0)
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: kubevirt_template
+
+short_description: Manage KubeVirt templates
+
+description:
+    - Use Openshift Python SDK to manage the state of KubeVirt templates.
+
+version_added: "2.8"
+
+author: KubeVirt Team (@kubevirt)
+
+options:
+    state:
+        description:
+            - Set the to be either I(present) or I(absent).
+        default: "present"
+        choices:
+            - present
+            - absent
+        type: str
+    name:
+        description:
+            - Name of the template.
+        required: true
+        type: str
+    namespace:
+        description:
+            - Namespace where the template exists.
+        required: true
+        type: str
+    objects:
+        description:
+            - List of any valid API objects, such as a I(DeploymentConfig), I(Service), etc. The object will be created exactly as defined here, with any parameter values substituted in prior to creation. The definition of these objects can reference parameters defined earlier.
+        type: list
+    vms:
```

Comment:
I wonder why we need to differentiate `vms`? Why not to keep everything in objects?

Reply:
So we can have it flat. It can be in objects, but then the VM spec won't be falt, but nested same as if you use API directly.

Reply:
Can we iterate over and find `known` objects and check simplified? We could use `handlers` for each object type and start with `vms` and add other kinds later.